x-pack/plugins/cases/public/components/cases_context/index.tsx

```diff
@@ -58,11 +75,17 @@ export const CasesProvider: React.FC<{ value: CasesContextProps }> = ({
   }, [appTitle, appId, userCanCrud]);
 
   return isCasesContextValue(value) ? (
-    <CasesContext.Provider value={value}>{children}</CasesContext.Provider>
+    <CasesContext.Provider value={value}>
```

Comment:
Will the `children` rerender if the state change?

Reply:
only the components that use that context.

Reply:
Hi @cnasikas , no, after [this change](https://github.com/elastic/kibana/pull/125505#discussion_r807876696) only the CasesContextUI will be updated on a state change, the context will not notice.

Reply:
I see! Thank you @semd! I was worried about that.